### PR TITLE
Add support for Google CAS v1

### LINKED ISSUE #607: Move to Google CAS public API

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds support for Google Certificate Authority Service v1.

Fixes #607

### CODE REVIEW

File: cas/apiv1/options.go [hunk 1]
```diff
@@ -45,10 +45,15 @@ type Options struct {
 	// KeyManager is the KMS used to generate keys in SoftCAS.
 	KeyManager kms.KeyManager `json:"-"`
 
-	// Project and Location are parameters used in CloudCAS to create a new
-	// certificate authority.
-	Project  string `json:"-"`
-	Location string `json:"-"`
+	// Project, Location, CaPool and GCSBucket are parameters used in CloudCAS
+	// to create a new certificate authority. If a CaPool does not exists it
```
Comment (dopey) [inline cas/apiv1/options.go:49]: does not exist

File: cas/cloudcas/cloudcas.go [hunk 2]
```diff
@@ -62,12 +67,22 @@ var revocationCodeMap = map[int]pb.RevocationReason{
 	10: pb.RevocationReason_ATTRIBUTE_AUTHORITY_COMPROMISE,
 }
 
+// caPoolTierMap contains the map between apv1.Options.Tier and the pb type.
```
Comment (dopey) [inline cas/cloudcas/cloudcas.go:70]: apiv1

**Review by dopey — APPROVED** [review]
lgtm. Couple of documentation grammar fixes in comments.

File: cas/cloudcas/cloudcas.go (hunk 2 above)
Comment (maraino) [reply cas/cloudcas/cloudcas.go]: Fixed in 35e6cc2

File: cas/apiv1/options.go (hunk 1 above)
Comment (maraino) [reply cas/apiv1/options.go]: Fixed in 35e6cc2

**Pull request merged**